sw/cheri/tests/plic_tests.hh

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright lowRISC contributors.
+ * Licensed under the Apache License, Version 2.0, see LICENSE for details.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+#include <cheri.hh>
+#include <functional>
+#include <platform-plic.hh>
+#include <platform-uart.hh>
+#include "../../common/defs.h"
+#include "../common/uart-utils.hh"
+#include "../common/asm.hh"
+#include "../common/sonata-devices.hh"
+#include "test_runner.hh"
+
+#include <array>
+ 
+struct PlicTest{
+  CapRoot root; 
+  PLIC::SonataPlic *plic;
+  UartPtr console;
+  uint32_t error_count = 0;
+
+  size_t instance = 0;
+  PLIC::Interrupts plic_irq_id ;
```

Comment:
Weird spacing again

Reply:
I rebased this PR and applied clang-format, should be fixed now